src/python/grpcio/grpc/_cython/_cygrpc/records.pyx.pxi

```diff
@@ -240,20 +252,27 @@ cdef class ByteBuffer:
     cdef size_t data_slice_length
     cdef void *data_slice_pointer
     if self.c_byte_buffer != NULL:
-      grpc_byte_buffer_reader_init(&reader, self.c_byte_buffer)
+      with nogil:
+        grpc_byte_buffer_reader_init(&reader, self.c_byte_buffer)
       result = b""
```

Comment:
Outside the scope of this change, but isn't this quadratic in time? I suppose we don't care until we have performance metrics telling us it's a problem?